src/mame/drivers/minitel_2_rpic.cpp

```diff
@@ -60,28 +72,32 @@
 
 #include "logmacro.h"
 
-// IO expander latch usage definitions
-enum
-{
-	CTRL_REG_MCBC  = 0x01,
-	CTRL_REG_DTMF  = 0x02,
-	CTRL_REG_CRTON = 0x08,
-	CTRL_REG_OPTO  = 0x10,
-	CTRL_REG_RELAY = 0x20
-};
+// 14174 Control register bits definition
+
+#define CTRL_REG_MCBC_BIT      0
+#define CTRL_REG_DTMF_BIT      1
+#define CTRL_REG_CRTON_BIT     3
+#define CTRL_REG_OPTO_BIT      4
+#define CTRL_REG_LINERELAY_BIT 5
 
 // 80C32 Port IO usage definitions
-enum
-{
-	PORT_1_KBSERIN = 0x01,
-	PORT_1_MDM_DCD = 0x02,
-	PORT_1_MDM_PRD = 0x04,
-	PORT_1_MDM_TXD = 0x08,
-	PORT_1_MDM_RTS = 0x10,
-	PORT_1_KBLOAD = 0x20,
-	PORT_1_SCL = 0x40,
-	PORT_1_SDA = 0x80
-};
+
+#define PORT_1_KBSERIN_BIT     0
+#define PORT_1_MDM_DCDn_BIT    1
+#define PORT_1_MDM_PRD_BIT     2
+#define PORT_1_MDM_TXD_BIT     3
+#define PORT_1_MDM_RTS_BIT     4
+#define PORT_1_KBLOAD_BIT      5
+#define PORT_1_SCL_BIT         6
+#define PORT_1_SDA_BIT         7
+
+#define PORT_3_SER_RXD_BIT     0
+#define PORT_3_SER_TXD_BIT     1
+#define PORT_3_SER_ZCO_BIT     2
+#define PORT_3_MDM_RXD_BIT     3
+#define PORT_3_SER_RDY_BIT     5
+
+#define MASK(bit) (0x01 << bit)
```

Comment:
Do you have a good reason for using macros?  In general, `enum` or `static constexpr` is preferred for constants, and `constexpr` functions are preferred for utility functions.  Macros provide worse compiler errors when there are issues, they produce less useful debug symbols, and function-like macros are error-prone due to the way substitution is done at preprocessing time (e.g. your macro here will behave unintuitively if the `bit` argument is an expression due to operator precedence rules).

Reply:
fixed. both defines and macro replaced.